### PR TITLE
ci(workflow): add cache to workflows using actions/setup-node

### PR DESCRIPTION
**Pull request by oscard0m**
## Description

- Add `cache` to workflows using `actions/setup-node`
- Replaces `bahmutov/npm-install` by `npm ci`. Now this GitHub Action is not necessary. Context: 
   - https://github.com/oscard0m/octoherd-script-add-cache-to-node-github-action/issues/13 and
   - https://github.com/probot/settings/issues/472#event-4979789613)

## Context

`setup-node` GitHub Action just released a new option to add cache to steps using it.

You can find the details here: https://github.blog/changelog/2021-07-02-github-actions-setup-node-now-supports-dependency-caching/

---

🤖 This PR has been generated automatically by [this octoherd script](https://github.com/oscard0m/octoherd-script-add-cache-to-node-github-action), feel free to run it in your GitHub user/org repositories! 💪🏾

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,8 +8,10 @@ jobs:
     steps:
       - uses: actions/checkout@v1
       - uses: actions/setup-node@v2
+        with:
+          cache: npm
       - uses: bahmutov/npm-install@v1
-      - run: 'npm run lint'
+      - run: "npm run lint"
```
Comment (oscard0m) [inline .github/workflows/ci.yml:14]:
```suggestion
      - run: 'npm run lint'
```

**Review by kaisermann — APPROVED** [review]
🎉  thanks!

**Comment by kaisermann** [comment]
@oscard0m Tests are failing because the action is expecting a npm lock, but the project uses yarn

**Comment by oscard0m** [comment]
> @oscard0m Tests are failing because the action is expecting a npm lock, but the project uses yarn

I would need some light here because it's not clear to me. Does this mean the steps should be running `yarn` instead of `npm ci` and, consequently, `cache: yarn` instead?

**Pull request closed without merging**